packages/core/src/provider/base.ts

```diff
@@ -109,12 +80,17 @@ export class SmartAccountProvider<
     | PublicErc4337Client<TTransport>
     | PublicErc4337Client<HttpTransport>;
 
-  constructor({
-    rpcProvider,
-    entryPointAddress,
-    chain,
-    opts,
-  }: SmartAccountProviderConfig<TTransport>) {
+  constructor(config: SmartAccountProviderConfig<TTransport>) {
+    try {
+      SmartAccountProviderConfigSchema<TTransport | HttpTransport>().parse(
+        config
+      );
+    } catch (e) {
+      throw new Error("invalid parameters passed to SmartAccountProvider");
+    }
```

Comment:
either rethrow the zod error or just run it without the try catch block

zod gives detailed messages about what part of schema validation fails which this error message does not capture